src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -272,35 +276,91 @@ CHIP_ERROR WiFiManager::GetNetworkStatistics(NetworkStatistics & stats) const
     return CHIP_NO_ERROR;
 }
 
-void WiFiManager::ScanResultClbk(NetEventCallback * cb)
+void WiFiManager::ScanResultHandler(uint8_t * data)
 {
-    VerifyOrReturn(Instance().mScanResultCallback != nullptr);
-    const struct wifi_scan_result * scanResult      = (const struct wifi_scan_result *) cb->info;
-    NetworkCommissioning::WiFiScanResponse response = ToScanResponse(scanResult);
-    Instance().mScanResultCallback(scanResult != nullptr ? &response : nullptr);
+    const struct wifi_scan_result * scanResult = reinterpret_cast<const struct wifi_scan_result *>(data);
+
+    if (Instance().mInternalScan &&
+        strcmp(reinterpret_cast<const char *>(scanResult->ssid), reinterpret_cast<const char *>(Instance().mWantedSSID.data())) ==
```

Comment:
How do we know it's null-terminated. Shouldn't we rather use `ssid_length`? Note that `ByteSpan` has `data_equals` method to ease such comparison.

Reply:
Good catch, thanks.